### PR TITLE
Update alfa parser

### PR DESCRIPTION
**Pull request by dnkolegov**

### CODE REVIEW

File: aule/generated/alfaAST.py [hunk 1]
```diff
@@ -67,66 +67,39 @@ def __init__(self, name=None, body=None, *args, **kwargs):
...
+        self.target = target
         self.rules = rules
-        self.events = events
         self.modifiers = modifiers    
     
     __mangled__ = {}
 
 class RuleDeclaration(Statement):
-    def __init__(self, name=None, effect=None, targetStatement=None, conditionStatement=None, events=None, *args, **kwargs):
+    def __init__(self, name=None, effect=None, target=None, events=None, *args, **kwargs):
```
Comment (yalegko) [inline aule/generated/alfaAST.py]: `events` should also be deleted

File: aule/resources/idl/alfa.idl [hunk 2]
```diff
@@ -19,46 +19,28 @@ interface NameSpaceDeclaration <: Statement {
...
     modifiers: [Keywords];
 }
 
 interface RuleDeclaration <: Statement {
     name: str;
     effect: EffectStatement;
-    targetStatement: TargetStatement;
-    conditionStatement: ConditionStatement;
+    target: TargetStatement;
     events: [Event];
```
Comment (yalegko) [inline aule/resources/idl/alfa.idl]: `event` field can't be parsed and should be deleted

File: aule/resources/idl/alfa.idl [hunk 3]
```diff
@@ -19,46 +19,28 @@ interface NameSpaceDeclaration <: Statement {
...
     name: str;
     algorithm: ApplyStatement;
-    targetStatement: TargetStatement;
-    conditionStatement: ConditionStatement;
+    target: TargetStatement;
     references: [Identifier];
     policies: [PolicyDeclaration];
     policysets: [PolicySetDeclaration];
-    events: [EventDeclaration];
     modifiers: [Keywords];
```
Comment (yalegko) [inline aule/resources/idl/alfa.idl]: We do not have a node called `Keyword`(s). Perhaps we should add it as a parent of `ExportKeyword`

File: aule/visitors/alfaASTVisitor.py [hunk 4]
```diff
@@ -13,14 +13,10 @@ def visitEffectStatement(self, ctx):
...
-        return CallExpression(callee)
-
     def visitAttributeAccessExpression(self, ctx):
         if not ctx.attributeAccessExpression():
             return Identifier(ctx.ID().getText())
-        return AttributeAccessExpression(self.visit(ctx.attributeAccessExpression()), ctx.ID().getText() )
+        return AttributeAccessExpression(self.visit(ctx.attributeAccessExpression()), ctx.ID().getText())
 
     def visitAttributeValue(self, ctx):
         text = ctx.getText()
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py:22]: Below we have a comparison `if text.lower() == "true" or text.lower() == "false":` and then we create `LiteralBoolean(text)`

So we can have `LiteralBoolean("TrUe")` and others which seems not very convenient to use.

Need to change constructor call to lowercase also

File: aule/visitors/alfaASTVisitor.py [hunk 5]
```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
...
 
     def visitTargetUnaryExpression(self, ctx):
         left = self.visit(ctx.targetExpression())
         return UnaryExpression("NOT", left)
 
-    def visitTargetBaseExpression(self, ctx):
-        left = self.visit(ctx.targetExpression(0))
-        right = self.visit(ctx.targetExpression(1))
+    def visitBooleanExpression(self, ctx):
+        return LiteralBoolean(ctx.getText())
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: Same here. I Suppose we can get bunch of quite tricky boolean literals w/o `.lower()`

File: aule/visitors/alfaASTVisitor.py [hunk 6]
```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
...
+    def visitTargetLogicalExpression(self, ctx):
         left = self.visit(ctx.targetExpression(0))
         right = self.visit(ctx.targetExpression(1))
-        return LogicalExpression("AND", left, right)
-
-    def visitTargetOrExpression(self, ctx):
-        left = self.visit(ctx.targetExpression(0))
-        right = self.visit(ctx.targetExpression(1))
-        return LogicalExpression("OR", left, right)
+        operator = ctx.logicalOperator().getText()
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: Need `.upper()` here to be same with `IN` and `NOT`

File: aule/visitors/alfaASTVisitor.py [hunk 7]
```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
...
+    def visitTargetAtom(self, ctx):
+        return self.visit(ctx.children[0])
+
     def visitTargetAnyExpression(self, ctx):
         left = self.visit(ctx.children[0].children[1])
         right = self.visit(ctx.children[0].children[3])
         return AnyExpression(left, right)
 
+    def visitTargetBooleanExpression(self, ctx):
+        return LiteralBoolean(ctx.getText())
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: `.lower()`

File: aule/visitors/alfaASTVisitor.py [hunk 8]
```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
     def visitBinaryOperator(self, ctx):
         return ctx.getText()
 
+    def visitSetOperator(self, ctx):
+        return ctx.getText()
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: Need `.upper()` here to be same with `IN` and `NOT`

File: aule/visitors/alfaASTVisitor.py [hunk 9]
```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
...
+
+    def visitLogicalOperator(self, ctx):
+        return ctx.getText()
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: Need `.upper()` here to be same with `IN` and `NOT`

**Review by yalegko — CHANGES_REQUESTED** [review]

File: aule/generated/alfaAST.py (hunk 1 above)
Comment (dnkolegov) [reply aule/generated/alfaAST.py]: Done

File: aule/resources/idl/alfa.idl (hunk 3 above)
Comment (dnkolegov) [reply aule/resources/idl/alfa.idl]: Done

File: aule/visitors/alfaASTVisitor.py (hunk 8 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: done

File: aule/visitors/alfaASTVisitor.py (hunk 9 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: Done

File: aule/visitors/alfaASTVisitor.py (hunk 6 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: Done

File: aule/visitors/alfaASTVisitor.py (hunk 7 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: Done

File: aule/visitors/alfaASTVisitor.py (hunk 5 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: Done

File: aule/visitors/alfaASTVisitor.py (hunk 4 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py:22]: Done

File: aule/codegen/generators/lua/alfa2lua.py [hunk 10]
```diff
@@ -71,6 +69,7 @@ def AttributeAccessExpression(self, node) -> str:
         return name
 
     def BinaryExpression(self, node) -> str:
+        print(node.operator)
```
Comment (yalegko) [inline aule/codegen/generators/lua/alfa2lua.py]: Remove debug output

File: aule/visitors/alfaASTVisitor.py [hunk 11]
```diff
@@ -61,7 +61,7 @@ def visitTargetStatement(self, ctx):
     def visitTargetLogicalExpression(self, ctx):
         left = self.visit(ctx.targetExpression(0))
         right = self.visit(ctx.targetExpression(1))
-        operator = ctx.logicalOperator().getText().upper()
+        operator = ctx.logicalOperator().getText().lower()
         return LogicalExpression(operator, left, right)
 
     def visitTargetUnaryExpression(self, ctx):
```
Comment (yalegko) [inline aule/visitors/alfaASTVisitor.py]: `69 return UnaryExpression("NOT", left)`

Now `NOT` is the only operator in uppercase. Need to change to 
`69 return UnaryExpression("not", left)`

**Review by yalegko — CHANGES_REQUESTED** [review]

File: aule/visitors/alfaASTVisitor.py (hunk 4 above)
Comment (yalegko) [reply aule/visitors/alfaASTVisitor.py:22]: Still there https://github.com/PositiveTechnologies/aule/blob/update-alfa-grammar/aule/visitors/alfaASTVisitor.py#L24

File: aule/codegen/generators/lua/alfa2lua.py (hunk 10 above)
Comment (dnkolegov) [reply aule/codegen/generators/lua/alfa2lua.py]: Ok

File: aule/visitors/alfaASTVisitor.py (hunk 11 above)
Comment (dnkolegov) [reply aule/visitors/alfaASTVisitor.py]: :-(

**Review by yalegko — APPROVED** [review]

**Comment by yalegko** [comment]
LGTM

**Pull request merged**